### PR TITLE
Feature/catch err create form

### PR DESCRIPTION
**Pull request by petia-latinova**

### CODE REVIEW

File: batch-nft-reveal/app/pages/create.tsx [hunk 1]
```diff
@@ -17,14 +18,20 @@ function CreatePage(): JSX.Element {
...
+      setError('')
+      await deployNFTCollection(args, library?.getSigner(), chainId)
+      .then((tx) => {
+        setDeployedContract({
+          address: tx.contractAddress,
+          txHash: tx.transactionHash,
+        })
+      })
+      .catch((ex) => {
+        ex.message ? setError(ex.message) : setError(ex)
```
Comment (imollov) [inline batch-nft-reveal/app/pages/create.tsx:34]: `await` + chained `then` and `catch` is not commonly used in React, not to mention this project.

File: batch-nft-reveal/app/pages/create.tsx [hunk 2]
```diff
@@ -17,25 +18,27 @@ function CreatePage(): JSX.Element {
...
-        )
-        setIsLoading(false)
-        setDeployedContract({
-          address: tx.contractAddress,
-          txHash: tx.transactionHash,
-        })
-      } catch (err) {
+        tx = await deployNFTCollection(args, library?.getSigner(), chainId)
+      } catch (ex) {
+        ex.message ? setError(ex.message) : setError(ex)
```
Comment (imollov) [inline batch-nft-reveal/app/pages/create.tsx:33]: Isn't `ex` expected to be an object?

Comment (petia-latinova) [reply batch-nft-reveal/app/pages/create.tsx:33]: No, we can't be sure about that. The errors from Metamask are objects for sure, but the deploy service have other code that can crash too and everything is wrapped in a single try-catch. The best I can do from here is if it's not a Metamask error to show the full message.

Comment (imollov) [reply batch-nft-reveal/app/pages/create.tsx:33]: My concern is setting `any` which could be an object to the error state which is expected to be `string`. Can't we just set a generic error message when that is the case?

**Review by imollov — APPROVED** [review]

**Pull request merged**